src/value.cc

```diff
@@ -301,9 +346,13 @@ Local<Value> ArrayToV8 (GITypeInfo *type_info, void* data, long length) {
 
     GIArgument value;
 
-    for (int i = 0; i < length; i++) {
+    for (int i = 0; (i < length || length == -1); i++) {
         void** pointer = (void**)((ulong)data + i * item_size);
         memcpy(&value, pointer, item_size);
+
+        if (length == -1 && isZero(value, item_type_info))
+            break;
+
```

Comment:
This seems a bit hard to read. Can we clearly separate zero-terminated and non-zero terminated cases?

Reply:
Does the current version look better? My other alternative is:

```c++
    if (length >= 0) {
        for (int i = 0; i < length; i++) {
            void** pointer = (void**)((ulong)data + i * item_size);
            memcpy(&value, pointer, item_size);

            Nan::Set(array, i, GIArgumentToV8(item_type_info, &value));
        }
    } else {
        for (int i = 0; ; i++) {
            void** pointer = (void**)((ulong)data + i * item_size);
            memcpy(&value, pointer, item_size);

            if (isZero(value, item_type_info))
                break;

            Nan::Set(array, i, GIArgumentToV8(item_type_info, &value));
        }
    }
```

But this feels silly.

Reply:
No, current version still feels off :|
The alternative is nice because it avoid unnecessary comparisons, but I don't think it matters enough. So anyway put it back however you want, if you could just assign `bool isZeroTerminated = length == -1;` so things are clear for whoever reads it.